### PR TITLE
Add internal order_routing_location_rule extension type

### PR DESCRIPTION
**Pull request by Keaton11**
### WHY are these changes introduced?

This PR adds the ``order_routing_location_rule`` extension type.

Related Issue: https://github.com/Shopify/shopify/issues/393386

### WHAT is this pull request doing?

This is one of [two changes](https://vault.shopify.io/page/Function-API-prototyping-guide~Dyet.md#registering-your-api-name) needed to register the ``order_routing_location_rule`` API name so that related functions can be deployed. 

### WHAT should reviewers focus on?

I based the template paths on those that were originally written for ``order_routing_ranker`` but these might need to be changed.

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by DuncanUszkay1** [comment]
>For the template path, does there need to be changes in https://github.com/Shopify/function-templates (adding the order routing directory and sample extensions) before this is merged?

I don't see much reason to add templates as nobody is allowed to upload these but us right? Seems like something we could circle back to post editions.

You might want to add the template first just so that you can tophat

**Review by DuncanUszkay1 — APPROVED** [review]
LGTM

**Review by luanpy — APPROVED** [review]
LGTM!

File: packages/app/src/cli/models/extensions/function-specifications/order_routing_location_rule.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {createFunctionSpec} from '../functions.js'
+
+const spec = createFunctionSpec({
+  identifier: 'order_routing_location_rule',
+  externalIdentifier: 'order_routing_location_rule',
+  externalName: 'Order routing location rule',
+  gated: true,
```
Comment (luanpy) [inline packages/app/src/cli/models/extensions/function-specifications/order_routing_location_rule.ts:7]: The [PR](https://github.com/Shopify/cli/pull/825/files#diff-b0e909dd71b245fc4d00358a78abb504fddc6bee0d9eb2068c927280ce14971b) where `public` was changed to `gated`

**Pull request merged**